src/Aspire.Hosting/ParameterResourceBuilderExtensions.cs

```diff
@@ -27,6 +27,54 @@ public static IResourceBuilder<ParameterResource> AddParameter(this IDistributed
         return builder.AddParameter(name, parameterDefault => GetParameterValue(builder.Configuration, name, parameterDefault), secret: secret);
     }
 
+    /// <summary>
+    /// Adds a parameter resource to the application, providing a default value to be used as a fallback.
+    /// </summary>
+    /// <param name="builder">Distributed application builder</param>
+    /// <param name="name">Name of parameter resource</param>
+    /// <param name="defaultValue">A string value that is used if the name is not found in the configuration</param>
+    /// <param name="secret">Optional flag indicating whether the parameter should be regarded as secret.</param>
+    /// <returns>Resource builder for the parameter.</returns>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters",
+                                                     Justification = "third parameters are mutually exclusive.")]
+    public static IResourceBuilder<ParameterResource> AddParameter(this IDistributedApplicationBuilder builder, string name, string defaultValue, bool secret = false)
+    {
+        // An alternate implementation is to use some ConstantParameterDefault implementation that always returns the default value.
+        // However, doing this causes a "default": {} to be written to the manifest, which is not valid.
+        // And note that we ignore parameterDefault in the callback, because it can never be non-null, and we want our own default.
+        return builder.AddParameter(name, parameterDefault => builder.Configuration[$"Parameters:{name}"] ?? defaultValue, secret: secret);
```

Comment:
I'm struggling with this one. Why is defaultValue a fallback instead of an override?

Reply:
It's the "default value". I.e. "if no one else supplied a value - use this one".

We would call it `string value`, if we wanted to always win.

Reply:
fallback value?

Reply:
Sounds like a continuation of your earlier conversation: https://github.com/dotnet/aspire/issues/4429#issuecomment-2158838761. And frankly, I'm a bit confused about the comment there that says that "code wins". To me, "config wins", and the code is a fall back, which is what my code is doing.

Reply:
In C#, when you have "optional arguments", the value that is used if you don't supply a value to the argument is called a "default vaule":

https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments#optional-arguments

> Each optional parameter has a default value as part of its definition. If no argument is sent for that parameter, the default value is used.

Reply:
> In C#, when you have "optional arguments", the value that is used if you don't supply a value to the argument is called a "default value"

Right, though this is quite different from our situation here, where the value can come from two completely different systems (config vs code). Here, I could see `fallbackValue` as a reasonable name, though I don't think `defaultValue` is so bad either.

Reply:
Started Discord thread to make it easier to discuss: https://discord.com/channels/732297728826277939/1281287950491193436